src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -879,15 +881,14 @@ public int GetSizeOfMajority()
 
         public async Task<int> WaitForReplicationAsync(int numberOfReplicasToWaitFor, TimeSpan waitForReplicasTimeout, string lastChangeVector)
         {
-            var numberOfSiblings = _destinations.Count;
+            var topology = GetClusterTopology();
+            var numberOfSiblings = _destinations.Select(x=>x.Url).Intersect(topology.AllNodes.Select(x=>x.Value)).Count();
```

Comment:
Let's avoid expensive Linq calls here.
Instead of doing this on each call, do it when we initialize the loader.